src/main/java/com/coveros/selenified/utilities/Property.java

```diff
@@ -0,0 +1,268 @@
+        return "true".equalsIgnoreCase(generatePDF);
+    }
+
+    /**
+     * Determines if we are supposed to zip up the results or not
+     *
+     * @return boolean: zip up the results or not
+     */
+    public static boolean packageResults() {
+        String packageResults = getProgramProperty(PACKAGE_RESULTS);
+        if (packageResults == null) {
+            return false;
+        }
+        if ("".equals(packageResults)) {
+            return true;
+        }
+        return "true".equalsIgnoreCase(packageResults);
+    }
+
+    /**
+     * Determines if a hub property is set. This could be to sauce, grid, or any other cloud tool.
+     * This should be provided with the protocol and address, but leaving out the /wd/hub
+     *
+     * @return boolean: is a hub location set
+     */
+    public static boolean isHubSet() {
+        String hub = getProgramProperty(HUB);
+        return hub != null && !"".equals(hub);
+    }
+
+    /**
+     * Retrieves the hub property if it is set. This could be to sauce, grid, or any other cloud tool.
+     * This should be provided with the protocol and address, but leaving out the /wd/hub
+     *
+     * @return String: the set hub address, null if none are set
+     */
+    public static String getHub() {
+        if (!isHubSet()) {
+            return null;
+        }
+        return getProgramProperty(HUB);
+    }
+
+    /**
+     * Determines if a proxy property is set. This could be to something local, or in the cloud.
+     * Provide the protocol, address, and port
+     *
+     * @return boolean: is a proxy set
+     */
+    public static boolean isProxySet() {
+        String proxy = getProgramProperty(PROXY);
+        return proxy != null && !"".equals(proxy);
+    }
+
+    /**
+     * Retrieves the proxy property if it is set. This could be to something local, or in the cloud.
+     * Provide the protocol, address, and port
+     *
+     * @return String: the set proxy address, null if none are set
+     */
+    public static String getProxy() {
+        if (!isProxySet()) {
+            return null;
+        }
+        return getProgramProperty(PROXY);
+    }
+
+
+    /**
+     * Obtains the application under test, as a URL. If the site was provided as
+     * a system property, that value will override whatever was set in the
+     * particular test suite. If no site was set, null will be returned, which
+     * will causes the tests to error out
+     *
+     * @param clazz   - the test suite class, used for making threadsafe storage of
+     *                application, allowing suites to have independent applications
+     *                under test, run at the same time
+     * @param context - the TestNG context associated with the test suite, used for
+     *                storing app url information
+     * @return String: the URL of the application under test
+     */
+    // TODO - should we align nomenclature? Either AppUrl (from CMD) or TestSite (from java code)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1135)